lobster_common/quaternion.py

```diff
@@ -26,30 +29,49 @@ def __init__(self, data: Union[List[float], Tuple[float, float, float, float], n
         if self._data.shape[0] != 4:
             raise InputDimensionError("A Quaternion needs an input array of length 4")
 
-    def numpy(self):
+    def numpy(self) -> np.ndarray:
         return self._data
 
     @property
-    def x(self):
+    def x(self) -> float:
         return self._data[X]
 
+    @x.setter
+    def x(self, value: float) -> None:
+        self._data[X] = value
+
     @property
-    def y(self):
+    def y(self) -> float:
         return self._data[Y]
 
+    @y.setter
+    def y(self, value: float) -> None:
+        self._data[Y] = value
+
     @property
     def z(self):
         return self._data[Z]
 
+    @z.setter
+    def z(self, value: float) -> None:
+        self._data[Z] = value
+
     @property
     def w(self):
         return self._data[W]
 
+    @w.setter
+    def w(self, value: float) -> None:
+        self._data[W] = value
+
     def __getitem__(self, key):
         return self._data[key]
 
     def __str__(self):
-        return f"Quaternion<{self._data}>"
+        return f"Quaternion<{self.x},{self.y},{self.z},{self.w}"
```

Comment:
Maybe nice to add xyzw before the numbers
```suggestion
        return f"Quaternion<x:{self.x},y:{self.y},z:{self.z},w:{self.w}"
```